vector/src/main/java/im/vector/home/FavoriteRoomFragment.kt

```diff
@@ -16,4 +17,14 @@ class FavoriteRoomFragment : BaseCommunicateHomeIndividualFragment() {
                 R.layout.adapter_item_room_view, true, onSelectRoomListener, invitationListener, moreActionListener)
         sectionView.setRooms(localRooms)
     }
+
+    override fun onAttach(context: Context) {
```

Comment:
I'd suggest leaving these functions in the super class and change the super class method impls to call an abstract method e.g.
```
    abstract fun getRoomFragment() : Fragment
    override fun onAttach(context: Context) {
        super.onAttach(context)
        registerListener?.onRegister(getRoomFragment(), this)
    }
```
which each of the subclasses implement e.g.
```
   override fun getRoomFragment() : Fragment {
       return CommunicateHomeFragment.ROOM_FRAGMENTS.FAVORITE
   }
```

Reply:
done